### PR TITLE
[7zip] Add C headers to includes

### PR DESCRIPTION
**Pull request by judemille**
The current 7zip port only adds the CPP headers to the includes. This PR adds the C headers, for consumers that need that.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download (not applicable)
- [x] The "supports" clause reflects platforms that may be fixed by this new version (not applicable)
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file. (no changes here)
- [x] Any patches that are no longer applied are deleted from the port's directory. (no changes)
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

The 7zip version 23.01 has been overridden. This can be changed if necessary.

### CODE REVIEW

**Comment by Osyotr** [comment]
> The 7zip version 23.01 has been overridden. This can be changed if necessary.

Please bump `port-version`. https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md#update-the-port-version-field-in-the-manifest-file-of-any-modified-ports

File: versions/7-/7zip.json [hunk 1]
```diff
@@ -1,7 +1,12 @@
 {
   "versions": [
     {
-      "git-tree": "770ce8dc829180bfcf2c396aa780474a87289f89",
+      "git-tree": "2f1fa323db9a88dd410e7d46ad651f7fe5cb620c",
+      "version-string": "23.01",
+      "port-version": 1
+    },
+    {
+      "git-tree": "dad5a164db0226aeca0477b350044c66324b2dca",
```
Comment (Osyotr) [inline versions/7-/7zip.json:9]: This is not supposed to change. I'm not sure what's the easiest way to fix this, but I would revert changes in this file and `baseline.json` and then call `vcpkg x-add-version 7zip`

Comment (judemille) [reply versions/7-/7zip.json]: Fixed that. Will squash commits if desired, though I'm not sure how that can be done in a manner that keeps vcpkg happy.

Comment (Osyotr) [reply versions/7-/7zip.json]: PRs in this repo are automatically squashed when merged. Note that these values are not commit hashes, but git tree hashes.

Comment (judemille) [reply versions/7-/7zip.json]: Ah, I see.

**Comment by judemille** [comment]
[CLA agreed to, begrudgingly.](https://github.com/microsoft/vcpkg/pull/34764#issuecomment-1793499580)

**Pull request merged**